audit_trail.go

```diff
@@ -0,0 +1,104 @@
+package tfe
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ AuditTrails = (*auditTrails)(nil)
+
+// AuditTrails describes all the audit trail related methods that the
+// Terraform Cloud API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/audit-trails.html
+type AuditTrails interface {
+	// List audit trails visible to the current user.
+	List(ctx context.Context, options AuditTrailListOptions) (*AuditTrailList, error)
+}
+
+// auditTrails implements AuditTrails.
+type auditTrails struct {
+	client *Client
+}
+
+// AuditTrail represents a Terraform Cloud audit trail.
+type AuditTrail struct {
```

Comment:
```suggestion
type AuditEvent struct {
```

Reply:
Hi @chrisarcand, thanks for looking at this. Sorry about the delay on my side this time!

Don't sweat about the renaming. I understand the importance of getting it right! I've changed the name from trail to event throughout the PR (apart from the API endpoint itself, and the link to the API doc).

Since raising this PR I've actually build a tool (using my fork) to store my org's audit events on our own S3 bucket. It works alright, but I noticed that I was only getting 1000 events because pagination wasn't working. I'm guessing it is because this API doesn't use the same JSON-API standard as the rest of them, and the json tags on the Pagination aren't allowing the raw response unmarshal correctly into the pagination attributes.

I ended up butchering the pagination struct a bit to make this work in my fork, but we obviously need a cleaner way to get it done in this repo.

I've included the butchered pagination struct in this PR, but I'm fairly sure it will break the rest of the APIs. I'm trying to think of the least-crufty way to make this work.